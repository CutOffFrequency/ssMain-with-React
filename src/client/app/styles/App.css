  * {
  text-align: center;
} .statusBanner {
  display: none;
} .loadTableButton-parentDiv {
  margin-top: 15px;
} table.data {
  margin: auto;
  width: 50%;
} table.data > * > * > * {
  border-style: solid;
  border-color: black;
  border-width: 1px;
}