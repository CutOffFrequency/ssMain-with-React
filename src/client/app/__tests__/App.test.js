import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import App from '../App';


it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
});

it('should match snapshot', () => {
    const tree = renderer.create(
        <App />
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

it('updates input value on change when value is a number in range', () => {
    const app = renderer.create(<App />);
    let tree = app.toJSON();
    // find child
    const selectDiv1 = tree.children.filter(child => child.type === 'div')[0];
    // find select
    const input1 = selectDiv1.children.filter(child => child.type === 'input')[0];
    expect(tree).toMatchSnapshot();
    let fakeEvent = {target: {value: "1"}}
    input1.props.onChange(fakeEvent);
    // re cache after changes
    tree = app.toJSON();
    expect(tree).toMatchSnapshot();
    const selectDiv2 = tree.children.filter(child => child.type === 'div')[0];
    const input2 = selectDiv2.children.filter(child => child.type === 'input')[0];
    expect(input2.props.value).toBe("1");
})

it('does not update input value on change when value is NaN', () => {
    const app = renderer.create(<App />);
    let tree = app.toJSON();
    // find child
    const selectDiv1 = tree.children.filter(child => child.type === 'div')[0];
    // find select
    const input1 = selectDiv1.children.filter(child => child.type === 'input')[0];
    expect(tree).toMatchSnapshot();
    // fake events
    let fakeEvent1 = {target: {value: "1"} }
    input1.props.onChange(fakeEvent1);
    let fakeEvent2 = {target: {value: "1x"} }
    input1.props.onChange(fakeEvent2);
    // recache
    tree = app.toJSON();
    expect(tree).toMatchSnapshot();
    const selectDiv2 = tree.children.filter(child => child.type === 'div')[0];
    const input2 = selectDiv2.children.filter(child => child.type === 'input')[0];
    expect(input2.props.value).toBe("1");
})

it('submits values in range and adds to select options', () => {
    const app = renderer.create(<App />);
    let tree = app.toJSON();
    // find child
    const selectDiv1 = tree.children.filter(child => child.type === 'div')[0];
    const selectDiv2 = tree.children.filter(child => child.type === 'div')[1];
    // find select
    const input1 = selectDiv1.children.filter(child => child.type === 'input')[0];
    const button1 = selectDiv1.children.filter(child => child.type === 'button')[0];
    expect(tree).toMatchSnapshot();
    // fake events
    let fakeEvent1 = {target: {value: "1"} }
    let fakeEvent2 = {target: {value: "2"} }
    input1.props.onChange(fakeEvent1);
    button1.props.onClick();
    input1.props.onChange(fakeEvent2);
    button1.props.onClick();
    tree = app.toJSON();
    expect(tree).toMatchSnapshot();
    const selectDiv3 = tree.children.filter(child => child.type === 'div')[1];
    const sel = selectDiv3.children.filter(child => child.type === 'select')[0];
    const child1 = sel.children.filter(child => child.type === 'option')[0];
    const child2 = sel.children.filter(child => child.type === 'option')[1];
    expect(child1.props.value).toBe("1");
    expect(child2.props.value).toBe("2");
})

it('does not submit a value that is out of range', () => {
    const app = renderer.create(<App />);
    let tree = app.toJSON();
    // find child
    const selectDiv1 = tree.children.filter(child => child.type === 'div')[0];
    const selectDiv2 = tree.children.filter(child => child.type === 'div')[1];
    // find select
    const input1 = selectDiv1.children.filter(child => child.type === 'input')[0];
    const button1 = selectDiv1.children.filter(child => child.type === 'button')[0];
    expect(tree).toMatchSnapshot();
    // fake events
    let fakeEvent1 = {target: {value: "0"} }
    let fakeEvent2 = {target: {value: "1"} }
    input1.props.onChange(fakeEvent1);
    button1.props.onClick();
    input1.props.onChange(fakeEvent2);
    button1.props.onClick();
    tree = app.toJSON();
    expect(tree).toMatchSnapshot();
    const selectDiv3 = tree.children.filter(child => child.type === 'div')[1];
    const sel = selectDiv3.children.filter(child => child.type === 'select')[0];
    const child = sel.children.filter(child => child.type === 'option')[0];
    expect(child.props.value).toBe("1");
})

it('changes selected value when select option is changed', () => {
    const app = renderer.create(<App />);
    let tree = app.toJSON();
    // find child
    const selectDiv1 = tree.children.filter(child => child.type === 'div')[0];
    const selectDiv2 = tree.children.filter(child => child.type === 'div')[1];
    // find select
    const input1 = selectDiv1.children.filter(child => child.type === 'input')[0];
    const button1 = selectDiv1.children.filter(child => child.type === 'button')[0];
    expect(tree).toMatchSnapshot();
    // fake events
    let fakeEvent1 = {target: {value: "1"} }
    let fakeEvent2 = {target: {value: "2"} }
    input1.props.onChange(fakeEvent1);
    button1.props.onClick();
    input1.props.onChange(fakeEvent2);
    button1.props.onClick();
    tree = app.toJSON();
    expect(tree).toMatchSnapshot();
    const selectDiv1_2 = tree.children.filter(child => child.type === 'div')[1];
    let sel = selectDiv1_2.children.filter(child => child.type === 'select')[0];
    const options = sel.children.filter(child => child.type === 'option');
    sel.props.onChange(fakeEvent1);
    tree = app.toJSON();
    expect(tree).toMatchSnapshot();
    let selectDiv1_3 = tree.children.filter(child => child.type === 'div')[1];
    sel = selectDiv1_3.children.filter(child => child.type === 'select')[0];
    expect(sel.props.value).toBe("1");
    sel.props.onChange(fakeEvent2);
    tree = app.toJSON();
    expect(tree).toMatchSnapshot();
    selectDiv1_3 = tree.children.filter(child => child.type === 'div')[1];
    sel = selectDiv1_3.children.filter(child => child.type === 'select')[0];
    expect(sel.props.value).toBe("2");
})