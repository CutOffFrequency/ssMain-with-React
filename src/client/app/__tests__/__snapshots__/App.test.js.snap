// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes selected value when select option is changed 1`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    />
  </div>
  <p>
    0 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`changes selected value when select option is changed 2`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
    </select>
  </div>
  <p>
    2 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`changes selected value when select option is changed 3`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value="1"
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
    </select>
  </div>
  <p>
    2 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`changes selected value when select option is changed 4`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value="2"
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
    </select>
  </div>
  <p>
    2 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`does not submit a value that is out of range 1`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    />
  </div>
  <p>
    0 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`does not submit a value that is out of range 2`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="1"
      >
        1
      </option>
    </select>
  </div>
  <p>
    1 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`does not update input value on change when value is NaN 1`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    />
  </div>
  <p>
    0 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`does not update input value on change when value is NaN 2`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value="1"
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    />
  </div>
  <p>
    0 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`should match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    />
  </div>
  <p>
    0 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`submits values in range and adds to select options 1`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    />
  </div>
  <p>
    0 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`submits values in range and adds to select options 2`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
    </select>
  </div>
  <p>
    2 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`updates input value on change when value is a number in range 1`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value=""
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    />
  </div>
  <p>
    0 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;

exports[`updates input value on change when value is a number in range 2`] = `
<div
  className="App"
>
  <div
    className="acctInput-parentDiv"
  >
    <p
      className="acctInput p"
    >
      enter an account
    </p>
    <input
      className="acctInput input"
      onChange={[Function]}
      value="1"
    />
    <button
      className="acctInput button"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="acctSelect-parentDiv"
  >
    <p
      className="acctSelect p"
    >
      select an account
    </p>
    <select
      className="acctSelect select"
      onChange={[Function]}
      value=""
    />
  </div>
  <p>
    0 accts loaded
  </p>
  <div
    className="tableSelect-parentDiv"
  >
    <p
      className="tableSelect p"
    >
      select a table
    </p>
    <select
      className="tableSelect select"
      onChange={[Function]}
      value=""
    >
      <option
        value="Form"
      >
        Form
      </option>
      <option
        value="Autos on Save"
      >
        Autos on Save
      </option>
      <option
        value="Autos on Deliver"
      >
        Autos on Deliver
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Customer database"
      >
        Customer database
      </option>
      <option
        value="History"
      >
        History
      </option>
      <option
        value="Picklists"
      >
        Picklists
      </option>
      <option
        value="On-Call"
      >
        On-Call
      </option>
      <option
        value="Skiplists"
      >
        Skiplists
      </option>
      <option
        value="Contacts"
      >
        Contacts
      </option>
      <option
        value="Admin"
      >
        Admin
      </option>
      <option
        value="Abend"
      >
        Abend
      </option>
      <option
        value="Help Topics"
      >
        Help Topics
      </option>
    </select>
  </div>
  <div
    className="statusBanner-parentDiv"
  >
    <div
      className="statusBanner "
    >
      
      <button
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
  <div
    className="loadTableButton-parentDiv"
  >
    <button
      className="loadTableButton button"
      onClick={[Function]}
    >
      load table
    </button>
  </div>
</div>
`;
